provider-ci/internal/pkg/templates/bridged-provider/Makefile

```
@@ -115,8 +115,23 @@ lint_provider: provider
 provider: tfgen install_plugins
 	(cd provider && go build $(PULUMI_PROVIDER_BUILD_PARALLELISM) -o $(WORKING_DIR)/bin/$(PROVIDER) -ldflags "-X $(PROJECT)/$(VERSION_PATH)=$(VERSION)#{{if .Config.providerVersion}}# -X #{{ .Config.providerVersion }}#=$(VERSION)#{{end}}#" $(PROJECT)/$(PROVIDER_PATH)/cmd/$(PROVIDER))
 
+#{{- if .Config.extraTests }}#
 test:
+	cd provider/shim && go test -v -coverprofile="coverage.txt" .
 	cd examples && go test -v -tags=all -parallel $(TESTPARALLELISM) -timeout 2h
+test.upgrade::
+	cd examples && go test -v -tags=all -run TestProviderUpgrade -timeout 2h
+
+test.upgrade.quick::
+	cd examples && go test -v -tags=all -run TestProviderUpgrade/Quick
+
+test.upgrade.record::
+	cd examples && PULUMI_ACCEPT=true go test -v -tags all -run TestProviderUpgrade -timeout 2h
+#{{- else }}#
+
+test:
+	cd examples && go test -v -tags=all -parallel $(TESTPARALLELISM) -timeout 2h
+#{{- end }}#
```

Can we avoid repeating the `test` target?
```suggestion
test:
#{{- if .Config.extraTests }}#
        # Run the shim test before the examples test, since we assume it is faster.
	cd provider/shim && go test -v -coverprofile="coverage.txt" .
#{{- end }}#
	cd examples && go test -v -tags=all -parallel $(TESTPARALLELISM) -timeout 2h
#{{- if .Config.extraTests }}#
test.upgrade::
	cd examples && go test -v -tags=all -run TestProviderUpgrade -timeout 2h

test.upgrade.quick::
	cd examples && go test -v -tags=all -run TestProviderUpgrade/Quick

test.upgrade.record::
	cd examples && PULUMI_ACCEPT=true go test -v -tags all -run TestProviderUpgrade -timeout 2h
#{{- end }}#
```